pandas/core/generic.py

```diff
@@ -10750,9 +10750,48 @@ def transform(self, func, *args, **kwargs):
         1  1.000000   2.718282
         2  1.414214   7.389056
         """
-        result = self.agg(func, *args, **kwargs)
-        if is_scalar(result) or len(result) != len(self):
-            raise ValueError("transforms cannot produce aggregated results")
+        raise NotImplementedError
+
+    def _transform(self, func, *args, **kwargs):
```

Comment:
rather than adding *anything* to the core files, do it in pandas/core/aggregation.py